pkgs/by-name/if/ifstate/package.nix

```
@@ -0,0 +1,78 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitea,
+  iproute2,
+  libbpf,
+  nixosTests,
+  withBpf ? false,
+  withConfigValidation ? true,
+  withShellColor ? false,
+  withWireguard ? true,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "ifstate";
+  version = "2.0.0rc4";
+  pyproject = true;
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "liske";
+    repo = "ifstate";
+    rev = version;
+    hash = "sha256-vc/srVA5ZqAH5112I9mlOWbpbCvZGvMHTup0GHym8EI=";
+  };
+
+  postPatch = ''
+    substituteInPlace libifstate/routing/__init__.py \
+      --replace-fail '/usr/share/iproute2' '${iproute2}/share/iproute2'
+  ''
+  + lib.optionalString withBpf ''
+    substituteInPlace libifstate/bpf/ctypes.py \
+      --replace-fail 'libbpf.so.1' '${libbpf}/lib/libbpf.so.1'
+  '';
+
+  build-system = with python3Packages; [
+    setuptools
+  ];
+
+  dependencies =
+    with python3Packages;
+    [
+      pyroute2
+      pyyaml
+      setproctitle
+    ]
+    ++ lib.optional withConfigValidation jsonschema
+    ++ lib.optional withShellColor pygments
+    ++ lib.optional withWireguard wgnlpy;
+
+  pythonRemoveDeps = lib.optional (!withConfigValidation) "jsonschema";
+
+  # does not has unit tests
```

```suggestion
  # has no unit tests
```
or "does not have any unit tests"